Doc/c-api/list.rst

```diff
@@ -112,6 +123,17 @@ List Objects
    to ``list.append(item)``.
 
 
+.. c:function:: int PyList_Extend(PyObject *self, PyObject *iterable)
+
+   Extend *list* with the contents of *iterable*. Similar to:
+   ``PyList_SetSlice(L, PY_SSIZE_T_MAX, PY_SSIZE_T_MAX, iterable)``.
```

Comment:
```suggestion
   ``PyList_SetSlice(list, PY_SSIZE_T_MAX, PY_SSIZE_T_MAX, iterable)``.
```

Reply:
Oh right, fixed.

Reply:
It has the same effect except that they raise different exceptions if the first argument is not a list.

Why is this? Why not simply raise a SystemError like in most of other Concrete Objects Layer C API?

Reply:
I would like to ask the question differently: why do other functions raise SystemError instead of TypeError?

In Python, TypeError is raised:

```
>>> list.extend("abc", "def")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'extend' for 'list' objects doesn't apply to a 'str' object

>>> list.clear("abc")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'clear' for 'list' objects doesn't apply to a 'str' object
```

Reply:
Because the simplest alternative to SystemError is no check at all. And an undefined behavior that often is manifested as a crash is the result. SystemError is a signal that the C API is used improperly. No correct program should get a SystemError.

1. At the low level: no checks, an undefined behavior, crash.
2. A higher Concrete Objects Layer: SystemError.
3. At Abstract Objects Layer and in Python: TypeError.